codepipeline/codepipeline.tf

```diff
@@ -0,0 +1,228 @@
+      category = "Source"
+      name     = "Source"
+      owner    = "AWS"
+      provider = "CodeStarSourceConnection"
+      version  = "1"
+
+      output_artifacts = ["source_output"]
+
+      configuration = {
+        ConnectionArn    = aws_codestarconnections_connection.default.arn
+        FullRepositoryId = var.repo_name
+        BranchName       = var.repo_branch_name
+      }
+
+    }
+
+  }
+
+  stage {
+    name = "Build"
+
+    action {
+      name             = "Build"
+      category         = "Build"
+      owner            = "AWS"
+      provider         = "CodeBuild"
+      input_artifacts  = ["source_output"]
+      output_artifacts = ["build_output"]
+      version          = "1"
+
+      configuration = {
+        ProjectName = aws_codebuild_project.default.name
+      }
+    }
+  }
+}
+
+resource "aws_codebuild_project" "default" {
+  name         = "${var.name_prefix}-codebuild"
+  service_role = aws_iam_role.codebuild_role.arn
+
+  artifacts {
+    type = "CODEPIPELINE"
+  }
+
+  environment {
+    compute_type                = "BUILD_GENERAL1_SMALL"
+    image                       = "aws/codebuild/amazonlinux2-x86_64-standard:2.0"
+    type                        = "LINUX_CONTAINER"
+    image_pull_credentials_type = "CODEBUILD"
+  }
+
+  logs_config {
+    cloudwatch_logs {
+      status = "ENABLED"
+    }
+  }
+
+  source {
+    type = "CODEPIPELINE"
+  }
+}
+
+resource "aws_s3_bucket" "default" {
+  bucket = "${var.name_prefix}-bucket"
+  acl    = "private"
+}
+
+# NOTE: Auth with the GitHub must be completed in the AWS Console.
+resource "aws_codestarconnections_connection" "default" {
+  name          = "${var.name_prefix}-connection"
+  provider_type = "GitHub"
+}
+
+resource "aws_iam_role" "codepipeline_role" {
+  name = "${var.name_prefix}-codepipeline_role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "codepipeline.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "codepipeline_role_policy" {
+  name = "${var.name_prefix}-codepipeline_role_policy"
+  role = aws_iam_role.codepipeline_role.id
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect":"Allow",
+      "Action": [
+        "s3:GetObject",
+        "s3:GetObjectVersion",
+        "s3:GetBucketVersioning",
+        "s3:PutObject",
+        "s3:PutObjectAcl"
+      ],
+      "Resource": [
+        "${aws_s3_bucket.default.arn}",
+        "${aws_s3_bucket.default.arn}/*"
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "codestar-connections:UseConnection"
+      ],
+      "Resource": [
+        "${aws_codestarconnections_connection.default.arn}"
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "codebuild:BatchGetBuilds",
+        "codebuild:StartBuild"
+      ],
+      "Resource": "*"
```

Comment:
This isn't a huge security risk since this policy should, in theory, only be used by CodePipeline, but it's generally a good idea to restrict access to only the resources that need to be accessed.  In this case, you would want to use the [arn for the CodeBuild project](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awscodebuild.html#awscodebuild-project).

Reply:
Changed to the CodeBuild project arn